influxql/ast.go

```diff
@@ -876,6 +906,48 @@ func (s *SelectStatement) Validate(tr targetRequirement) error {
 		}
 	}
 
+	if err := s.validateDerivative(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SelectStatement) validateDerivative() error {
+	if !s.HasDerivative() {
+		return nil
+	}
+
+	// If a derivative is requested, it must be the only field in the query. We don't support
+	// multiple fields in combination w/ derivaties yet.
+	if len(s.Fields) != 1 {
+		return fmt.Errorf("derivative cannot be used with other fields")
+	}
+
+	aggr := s.FunctionCalls()
+	if len(aggr) != 1 {
+		return fmt.Errorf("derivative cannot be used with other fields")
+	}
+
+	// Derivative requires two arguments
+	derivativeCall := aggr[0]
+	if len(derivativeCall.Args) != 2 {
+		return fmt.Errorf("derivative requires a field and duration argument")
```

Comment:
@pauldix 
- Can the derivative duration differ from the group by time?  If so, what does that do?
- For raw queries, shouldn't it be between every point?

The requests I remember were for derivative between every point on raw queries, derivative of first and last point in a bucket, and derivative of last point in one bucket to first point in next bucket.  But, it's been a while, so maybe I'm remember wrong.